spec/compiler/lexer/lexer_spec.cr

```diff
@@ -558,4 +558,18 @@ describe "Lexer" do
   assert_syntax_error %("\\x1z"), "invalid hex escape"
 
   assert_syntax_error %("hi\\)
+
+  # CVE-2021-42574
+  describe "trojan source" do
```

Comment:
Might want to have extra specs that ensure their escape sequences are still permitted in Crystal source code.

Reply:
Not sure if that's necessary because parsing escape sequences has nothing to do with that. But I suppose it doesn't hurt.